lib/RexStanSettings.php

```diff
@@ -116,4 +116,73 @@ public static function createForm()
 
         return $form;
     }
+
+    /**
+     * @return array<string, string>
+     */
+    public static function getSettings()
+    {
+        $addon = rex_addon::get('rexstan');
+
+        $scanTargets = [];
+        $config_scanTargets = explode('|', trim(strval($addon->getConfig('addons')), '|'));
+        foreach ($config_scanTargets as $scanpath) {
+            foreach (rex_addon::getAvailableAddons() as $availableAddon) {
+                if (pathinfo($scanpath)['basename'] === $availableAddon->getName()) {
+                    $scanTargets[] = $availableAddon->getName();
+                    break;
+                }
+                if (isset(pathinfo($scanpath)['dirname']) && pathinfo(pathinfo($scanpath)['dirname'])['basename'] === 'developer') {
+                    if (pathinfo($scanpath)['basename'] === 'modules' || pathinfo($scanpath)['basename'] === 'templates') {
+                        $scanTargets[] = 'developer:' . pathinfo($scanpath)['basename'];
+                        break;
+                    }
+                }
+            }
+        }
+
+        $extensions = [];
+        $config_extensions = explode('|', trim(strval($addon->getConfig('extensions')), '|'));
+        foreach ($config_extensions as $extpath) {
+            foreach (self::$phpstanExtensions as $label => $path) {
+                if (pathinfo($extpath)['basename'] === pathinfo($path)['basename']) {
+                    $extensions[] = $label;
+                    break;
+                }
+            }
+        }
+
+        $settings = [];
+        $settings['Level'] = strval($addon->getConfig('level'));
+        $settings['AddOns'] = implode(', ', $scanTargets);
+        $settings['Extensions'] = implode(', ', $extensions);
+        $settings['PHP-Version'] = self::$phpVersionList[$addon->getConfig('phpversion')];
```

Comment:
schön wäre, wenn wir statt diesem array eine eigene klasse mit ordentlichen properties und typen declariert wäre (z.b. `RexStanConfigVO`)

Reply:
Meinst Du eine Klasse mit getter+setter oder nur die Variablen mit Typen kapseln!?

Und warum ...VO ? Für was steht das VO :)

Reply:
Eine klasse nur mit getter und die werte kommen via __construct() rein. VO == ValueObject

Reply:
Habe jetzt die Settings in der Klasse `RexStanConfigVO` umgesetzt.

Guckst Du mal @staabm